Lib/functools.py

```diff
@@ -16,7 +16,8 @@
 
 from abc import get_cache_token
 from collections import namedtuple
-# import types, weakref  # Deferred to single_dispatch()
+import types
```

Comment:
We were already importing `types` for `GenericAlias`, so I cleaned up the function-local import.